pom.xml

```diff
@@ -0,0 +1,135 @@
+        <project.build.sourceEncoding>${project.encoding}</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>${project.encoding}</project.reporting.outputEncoding>
+
+        <keycloak.version>4.8.3.Final</keycloak.version>
+        <commons-io.version>2.6</commons-io.version>
+        <spring-boot.version>2.1.4.RELEASE</spring-boot.version>
+        <junit.version>4.12</junit.version>
+        <mockito.version>1.10.19</mockito.version>
+    </properties>
+
+    <scm>
+        <connection>scm:git:https://github.com/Alfresco/alfresco-rest-authn-java-client.git</connection>
+        <developerConnection>scm:git:https://github.com/Alfresco/alfresco-rest-authn-java-client.git</developerConnection>
+        <url>https://github.com/Alfresco/alfresco-rest-authn-java-client</url>
+        <tag>HEAD</tag>
+    </scm>
+
+    <repositories>
+        <repository>
+            <id>alfresco-public</id>
+            <url>https://artifacts.alfresco.com/nexus/content/groups/public</url>
+        </repository>
+        <repository>
+            <id>alfresco-internal</id>
+            <url>https://artifacts.alfresco.com/nexus/content/groups/internal</url>
+        </repository>
+    </repositories>
+
+    <pluginRepositories>
+        <pluginRepository>
+            <id>alfresco-public</id>
+            <url>https://artifacts.alfresco.com/nexus/content/groups/public</url>
+        </pluginRepository>
+    </pluginRepositories>
+
+    <distributionManagement>
+        <repository>
+            <id>alfresco-internal</id>
+            <url>https://artifacts.alfresco.com/nexus/content/repositories/releases</url>
+        </repository>
+        <snapshotRepository>
+            <id>alfresco-internal-snapshots</id>
+            <url>https://artifacts.alfresco.com/nexus/content/repositories/snapshots</url>
+        </snapshotRepository>
+    </distributionManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-adapter-core</artifactId>
+            <version>${keycloak.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-authz-client</artifactId>
+            <version>${keycloak.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>${commons-io.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter</artifactId>
+            <version>${spring-boot.version}</version>
+            <scope>compile</scope>
+            <optional>true</optional>
```

Comment:
What was the reason for an optional dependency rather than sub-modules?

Reply:
To make the developer life easier, the AuthConfigBuilder has an overloaded constructor that takes a Spring `Environment` object.    
In a Spring Env, our common practice is to override the properties in the `application.properties` file, therefore, all needed is to Autowire the Environment bean and pass that to the AuthnConfigBuilder. 
So, the `optional` dependency is needed in order to compile the code. Any client that wants to use the `Builder(Environment environment)` constructor, must have Spring as the dependency.